File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/examples-tables/examples_tables.cs
using Reqnroll;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CucumberMessages.CompatibilityTests.CCK.examples_tables
{
    [Binding]
    internal class Example_tables
    {
        private int _count;
        private int _friends;

        [Given("there are {int} cucumbers")]
        public void GivenThereAreCucumbers(int p0)
        {
            _count = p0;
        }

        [Given("there are {int} friends")]
        public void GivenThereAreFriends(int p0)
        {
            _friends = p0;
        }

        [When("I eat {int} cucumbers")]
        public void WhenIEatCucumbers(int p0)
        {
            _count -= p0;
        }

        [Then("I should have {int} cucumbers")]
        public void ThenIShouldHaveCucumbers(int expectedCount)
        {
            if (!expectedCount.Equals(_count))
                throw new ApplicationException($"Count: {_count} not equal to ExpectedCount: {expectedCount}");
        }

        [Then("each person can eat {int} cucumbers")]
        public void ThenEachPersonCanEatCucumbers(int p0)
        {
            var share = Math.Floor((double)_count / (1 + _friends));

            if (p0 != share)
                throw new ApplicationException($"Incorrect calculated share: {share} not equal to Expected share: {p0}");
        }
    }
}

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/examples-tables/examples_tables.ts
import assert from 'node:assert'
import { Given, When, Then } from '@cucumber/fake-cucumber'

Given('there are {int} cucumbers', function (initialCount) {
  this.count = initialCount
})

Given('there are {int} friends', function (initialFriends) {
  this.friends = initialFriends
})

When('I eat {int} cucumbers', function (eatCount) {
  this.count -= eatCount
})

Then('I should have {int} cucumbers', function (expectedCount) {
  assert.strictEqual(this.count, expectedCount)
})

Then('each person can eat {int} cucumbers', function (expectedShare) {
  let share = Math.floor(this.count / (1 + this.friends));
  assert.strictEqual(share, expectedShare)
})
